### PR TITLE
Avoid importing from pandas at _libs files

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Something that brought at to me at first [here](https://github.com/pandas-dev/pandas/pull/30395#discussion_r363101229)

---

cc @jbrockmendel

### CODE REVIEW

**Comment by WillAyd** [comment]
What is the benefit of doing this? I think I prefer it as is so we have only one way of testing for 32 bit

**Comment by MomIsBestFriend** [comment]
@WillAyd TBH I don't really know, all I do know is that one time I imported ```pandas.compat``` at the top, and I was requested to change it.

**Comment by jbrockmendel** [comment]
@MomIsBestFriend is right that in general we try to avoid non-_libs imports in _libs, but that rule is not hard-and-fast, e.g. _libs.parsers is all over the place.  tslibs is the only place where im really adamant about this.

File: pandas/_libs/missing.pyx [hunk 1]
```diff
@@ -16,17 +17,12 @@ from pandas._libs.tslibs.nattype cimport (
...
-from pandas.compat import is_platform_32bit
-
-
 cdef:
     float64_t INF = <float64_t>np.inf
     float64_t NEGINF = -INF
 
     int64_t NPY_NAT = util.get_nat()
 
-    bint is_32bit = is_platform_32bit()
```
Comment (jbrockmendel) [inline pandas/_libs/missing.pyx]: can you do `bint is_32bit = platform.architecture()[0] == "32-bit"` and avoid re-computing at runtime?

**Comment by WillAyd** [comment]
Unless there is a performance improvement I'd be -1 on this change

**Comment by jreback** [comment]
what actually need is a *cython* version of compat that we can use; this is not what we are going for.

**Pull request closed without merging**